### PR TITLE
Add MAX32620 HIC and MAX32625MBED board

### PR DESCRIPTION
**Pull request by khariya**

### CODE REVIEW

File: records/hic_hal/max32620.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+common:
+    target:
+        - max32620
+    macros:
```
Comment (brianesquilona) [inline records/hic_hal/max32620.yaml]: Need to add a core here, for example:
`core:
        - Cortex-M4`

Comment (khariya) [reply records/hic_hal/max32620.yaml]: Do Cortex-M4 with and w/o FPU have same names in this file?
MAX32620 is based on Arm Cortex-M4 with FPU.

Comment (brianesquilona) [reply records/hic_hal/max32620.yaml]: Yes it should be M4F

Comment (khariya) [reply records/hic_hal/max32620.yaml]: Thanks for updating that!

**Pull request merged**